### DISCUSSION TITLE
DietPi-Software | Add evcc ☀️🚗

### DISCUSSION THREAD

**Discussion by naltatis** [MichaIng/DietPi#6909] [Software Request]
#### Formal software information
- Software title | evcc
- Short description | Open Source Solar Charging
- Official URL (if available) | https://evcc.io
- GitHub URL (if available) | https://github.com/evcc-io/evcc
- (Official) install documentation | https://docs.evcc.io/docs/installation/linux

#### Are there similar/alternative software titles available with DietPi-Software?
Home Automation

- Home Assistant
- OpenHAB

#### What makes your requested software better than the above solutions, if available?

evcc is focused on smart EV charging with own solar or cheap grid energy (e.g. Tibber). It's an integrated solution around this specific use case and in contrast to the above projects, not a general purpose home automation system. We provide integrations with these systems (as well as influx, grafana, mqtt, ...). So i think its a good addition to the current software ecosystem on dietpi.

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?

I guess the apt installation is the easiest way.

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
```
sudo apt install -y debian-keyring debian-archive-keyring apt-transport-https curl
curl -1sLf 'https://dl.cloudsmith.io/public/evcc/stable/setup.deb.sh' | sudo -E bash
sudo apt update
sudo apt install -y evcc
sudo systemctl start evcc
```

After the installation, evcc runs in demo mode at http://127.0.0.1:7070/.

Note: evcc does not have a complete web configuration right now. This will come in one of the next releases, together with the requirement to set an administrator password during setup.

For now, users have to create an `/etc/evcc.yaml` via our cli assistant `evcc configure` or according to our documentation. It would be nice if this is highlighted as a text message at the end of the dietpi installation process. Once our web config is available, I'll create a new issue/pr with updated instructions.

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.

Yes, I'm one of the evcc developers and have an interest in providing an easy installation for our users on as many platforms as possible.

**Comment by tolot27** [comment]
Has anyone run evcc on DietPi for a long time?
If yes on which device?
